ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -653,7 +653,7 @@ def add_host_ports(self):
                     ptf_if = PTF_FP_IFACE_TEMPLATE % host_ifindex
                     self.add_dut_if_to_docker(ptf_if, fp_port)
             else:
-                fp_port = self.duts_fp_ports[self.duts_name[0]][intf]
+                fp_port = self.duts_fp_ports[self.duts_name[0]][str(intf)]
```

Comment:
In `add_host_ports`, the type change from list to dictionary will break the code in the circumstances that `self._is_multi_duts` evaluates to `True`.

Reply:
Good catch - my testing was dealing with no host interfaces in a multi-dut environment and thus did not catch this. Committed a fix for this